### ISSUE TITLE
do not transfer userlogs for failed jobs

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#5388]
we must stop movin userlogs with ASO
nobody needs them, nobody really knows what they are, but very many users have this enabled in their configuration w/o even knowing it.
The few thousand lines of CMSSW stdout/err that we retrieve via the schedd have to suffice. If someone needs event-level info, they should learn how to write a proper file.

What happens now is that zilion of jobs crash soon or immediately, produce no output data, and yet overload ASO with those useless logs. Especially obnoxious are jobs who fail in first minutes and retry very quickly. We can likley reduce ASO load by a factor 2 at no cost here.

**Comment by belforte** [comment]
we should also make crab getlog pull files form schedd, not from T2
and if logs are gone after a week, so be it.

**Comment by PerilousApricot** [comment]
Preventing the logs disappearing is the reason many people transfer them,
for the trigger report. There will be a lot of unhappy users if the logs
just evaporate a week :/
-- 
It's dark in this basement.

**Comment by belforte** [comment]
first step could be to avoid moving around logs for failed jobs which lasted only 30sec.
Then we should maybe poll users ? On replacing full log with the shortened (4k lines !) version that goes to schedd as well, and make sure we zip them.
@PerilousApricot I see, but .. can't/shouldn't cmsRun provide the triggerreport information in a compact way ?

**Comment by belforte** [comment]
will not transfer log for failed jobs. Period.

**Comment by belforte** [comment]
there some flawed login in my commit which makes ASO stuck forever.

**Comment by belforte** [comment]
From @dciangot 
```
In cmscp.py I think there is a bug. The env fail_no_trans_log has to be
true to disable the transfers, not false like now.
```

**Comment by belforte** [comment]
After running a few test tasks I believe that there is nothing to do here !
There is some confusion in cmscp.py since when job fails and are retried and re-run at same site, local stagetout rightfully fails because output file is already there, yet some subscript exits with code 0 and cmscp says thl eat root file was staged out OK. Anyhow somehow the post job does not schedule any ASO transfer for failed jobs, so we are fine.
I suspect that this issue started based on some memory of how things were when we were injecting in ASO from the WN. Now that we do not do it, we simply pollute local stage out .../failed/...directory with files that nobody will ever read or deleted. I am inclined to simply skip cmscp whenever exit code is not 0. But given how fragile all of this appears to be. Better stick to "it works, don't touch it".
@dciangot can you please give your opinion ?

**Comment by belforte** [comment]
hmm.. PostJob logic is very hard for me to follow by reading. I am not sure why exactly ASO xfer was not attempted in my test task (where applciation ended with a recoverable error 8021).
Need to dif more before closing/forgetting

**Comment by belforte** [comment]
OK. I think I have districatd things with pdb.
PJ for first attempt (i.e. before entering wait_for_ASO loop)  goes through
https://github.com/dmwm/CRABServer/blob/eb658c899ac09a4da8ac1785733b0aa5e54f8064/src/python/TaskWorker/Actions/PostJob.py#L1820-L1823

and executes `RetryJob()` 
in both case where it returnes RECOVERAL_ERROR (=1) or FATAL_ERROR (=2) line
https://github.com/dmwm/CRABServer/blob/eb658c899ac09a4da8ac1785733b0aa5e54f8064/src/python/TaskWorker/Actions/PostJob.py#L1822
makes the PJ return at this point w/o executing any ASO related code.

Bottom line: ASO is only called in for jobs with exit code = 0.

**Comment by dciangot** [comment]
The point here was that the changes in cmscp caused:

> it looks like for some reason the job logs are not copied on the local tmp area although the transferlogs flag is true. This cause ASO transfers for the logs to fail and, as consequence, to make the whole task transfer process failing even if the real output file is actually transferred. See 181005_200805:pozzo_crab_DTHough for example.

So the use case you should test is when a job with transferlogs flag succeeded, and you'll see that it will failed in the postprocessing then.
That said, I have difficulties to follow all your consideration above - so I just hope to have clarified the point of this issue.

**Comment by belforte** [comment]
@dciangot AFAIU you are talking about the bug which I introduced, that's understood. If I were to change cmscp again I surelly need to be careful and propetly test. But here I was wondering whether we need to do anything at alll, i.e. whether the original use case (useless transfer of log files for failed jobs) really exists anymore. OTOH since it turned out that I was using the ASOless version of the PJ, maybe we need to revisit ?

**Comment by dciangot** [comment]
The logic should not change on asoless, since what to transfer is decided
on the pj totally. I think that we should stay as it is now. Am I missing
something?

Diego

Il giorno gio 1 nov 2018, 15:13 Stefano Belforte <notifications@github.com>
ha scritto:

**Comment by belforte** [comment]
I think you are right. Simply I do not understand the code at the moment so was not daring to draw conclusions.
